api/configs/middleware/__init__.py

```diff
@@ -171,7 +171,7 @@ def SQLALCHEMY_DATABASE_URI(self) -> str:
 
     RETRIEVAL_SERVICE_EXECUTORS: NonNegativeInt = Field(
         description="Number of processes for the retrieval service, default to CPU cores.",
-        default=os.cpu_count(),
+        default=os.cpu_count() or 8,
```

Comment:
why use 8 here?

Reply:
The `os.cpu_count()` returns `Optional[int]` which violates `mypy` type checks for NonNegativeInt. Adding an default int to pass the checks. In general most cases , `os.cpu_count()` does return valid int and it woun't fall to the default value.

Reply:
Changed the default value to 1. Solved.